api/scripts/send-invitations-to-sco-organizations.js

```diff
@@ -1,118 +1,75 @@
-    }
-    if (!email) {
-      if (require.main === module) process.stdout.write(`A line is missing a email for external id ${externalIdLowerCase}`);
-      return null;
-    }
-    const externalId = externalIdLowerCase.toUpperCase();
+const organizationRepository = require('../lib/infrastructure/repositories/organization-repository');
+const organizationInvitationRepository = require('../lib/infrastructure/repositories/organization-invitation-repository');
 
-    return { externalId, email: email.trim() };
-  }).filter((data) => !!data);
+async function getOrganizationByExternalId(externalId) {
+  return BookshelfOrganization
+    .where({ externalId })
+    .fetch({ require: true })
+    .then((organization) => bookshelfToDomainConverter.buildDomainObject(BookshelfOrganization, organization))
+    .catch((err) => {
+      if (err instanceof BookshelfOrganization.NotFoundError) {
+        throw new NotFoundError(`Organization not found for External ID ${externalId}`);
+      }
+      throw err;
+    });
 }
 
-async function sendInvitations({ accessToken, organizationsByExternalId, checkedData }) {
-  for (let i = 0; i < checkedData.length; i++) {
-    if (require.main === module) process.stdout.write(`\n${i + 1}/${checkedData.length} `);
-
-    const { externalId, email } = checkedData[i];
-    const organization = organizationsByExternalId[externalId];
-
-    if (organization && organization.id && email) {
-      await request(_buildPostOrganizationInvitationRequestObject(accessToken, organization, email));
-    }
-  }
+async function buildInvitation({ externalId, email }) {
+  const { id: organizationId } = await getOrganizationByExternalId(externalId);
+  return { organizationId, email };
 }
 
-function _buildAccessTokenRequestObject() {
-  return {
-    method: 'POST',
-    baseUrl,
-    url: '/api/token',
-    headers: {
-      'Content-Type': 'application/x-www-form-urlencoded',
-    },
-    form: {
-      grant_type: 'password',
-      username: process.env.PIXMASTER_EMAIL,
-      password: process.env.PIXMASTER_PASSWORD,
-    },
-    json: true,
-  };
+async function prepareDataForSending(objectsFromFile) {
+  return bluebird.mapSeries(objectsFromFile, ({ uai, email }) => {
+    return buildInvitation({ externalId: uai, email });
+  });
 }
 
-function _buildPostOrganizationInvitationRequestObject(accessToken, organization, email) {
-  return {
-    method: 'POST',
-    headers: {
-      authorization: `Bearer ${accessToken}`,
-    },
-    baseUrl,
-    url: `/api/organizations/${organization.id}/invitations`,
-    json: true,
-    body: {
-      data: {
-        type: 'organization-invitations',
-        attributes: {
-          email,
-        },
-      },
-    },
-  };
+async function sendJoinOrganizationInvitations(invitations) {
+  let nbr = 0;
+
+  return bluebird.mapSeries(invitations, ({ organizationId, email }) => {
+    if (require.main === module) {
+      process.stdout.write(`${++nbr}/${invitations.length}\r`);
+    }
```

Comment:
Un index est disponible en deuxième paramètre de la callback 😃 

```suggestion
  return bluebird.mapSeries(invitations, ({ organizationId, email }, index) => {
    if (require.main === module) {
      process.stdout.write(`${index}/${invitations.length}\r`);
    }
```

Reply:
Done